packages/legend-query-builder/src/components/__tests__/QueryBuilderExecution.test.tsx

```diff
@@ -0,0 +1,127 @@
+  string,
+  {
+    mappingPath: string;
+    runtimePath: string;
+    entities: Entity[];
+    rawLambda: { parameters?: object; body?: object };
+    expectedNumberOfParameter: number;
+    expectedNumberOfParameterValues: number;
+    executionResult: PlainObject<unknown>;
+  },
+];
+
+const QUERY_PROPERTY_PARAMETER_CASES: QueryPropertyParameterTestCase[] = [
+  [
+    'test query execution with parameters',
+    {
+      mappingPath: TEST_DATA_QueryExecution_ExecutionInput.mapping,
+      runtimePath: TEST_DATA_QueryExecution_ExecutionInput.runtime.runtime,
+      entities: TEST_DATA_QueryBuilder_QueryExecution_Entities,
+      rawLambda: TEST_DATA_QueryExecution_ExecutionInput.function,
+      expectedNumberOfParameter: 1,
+      expectedNumberOfParameterValues: 1,
+      executionResult: TEST_DATA_QueryExecution_ExecutionResult,
+    },
+  ],
+];
+
+describe(integrationTest(`test query execution with parameters`), () => {
+  test.each(QUERY_PROPERTY_PARAMETER_CASES)(
+    '%s',
+    async (
+      testName: QueryPropertyParameterTestCase[0],
+      testCase: QueryPropertyParameterTestCase[1],
+    ) => {
+      const {
+        mappingPath,
+        runtimePath,
+        entities,
+        rawLambda,
+        expectedNumberOfParameter,
+        expectedNumberOfParameterValues,
+      } = testCase;
+      const { queryBuilderState } = await TEST__setUpQueryBuilder(
+        entities,
+        stub_RawLambda(),
+        mappingPath,
+        runtimePath,
+      );
+      await act(async () => {
+        queryBuilderState.initializeWithQuery(
+          create_RawLambda(rawLambda.parameters, rawLambda.body),
+        );
+      });
+      expect(queryBuilderState.parametersState.parameterStates.length).toBe(
+        expectedNumberOfParameter,
+      );
+      const expectedParameterValues =
+        TEST_DATA_QueryExecution_ExecutionInput.parameterValues.map((p) => {
+          const parameterValue = new ParameterValue();
+          parameterValue.name = p.name;
+          parameterValue.value = p.value;
+          return parameterValue;
+        });
+      const multiplicityOne =
+        queryBuilderState.graphManagerState.graph.getTypicalMultiplicity(
+          TYPICAL_MULTIPLICITY_TYPE.ZEROONE,
+        );
+      queryBuilderState.parametersState.parameterStates.forEach(
+        (queryParamState) => {
+          queryParamState.value = new PrimitiveInstanceValue(
+            GenericTypeExplicitReference.create(
+              new GenericType(
+                queryBuilderState.graphManagerState.graph.getPrimitiveType(
+                  PRIMITIVE_TYPE.INTEGER,
+                ),
+              ),
+            ),
+            multiplicityOne,
+          );
+          (queryParamState.value as PrimitiveInstanceValue).values = [20];
+        },
+      );
+      const parameterValues =
+        queryBuilderState.resultState.buildExecutionParameterValues();
+      expect(parameterValues).toEqual(expectedParameterValues);
```

Comment:
how is the equal on this done? i.e i would expect them to fail since they are different objects.
Woudlnt it better to compare the serialized json from the parameter values of both?